numba/core/codegen.py

```diff
@@ -60,8 +62,369 @@ class _CFG(object):
+
+            # loop peel the block header, it needs a HTML border
+            fmtheader = ('<tr><td BGCOLOR="{}" BORDER="1" ALIGN="left" '
+                         'COLSPAN="{}">{}</td></tr>')
+            new_lines.append(fmtheader.format(cs['default'], col_span,
+                                              clean(sliced_lines[0].strip())))
+
+            # process rest of block creating the table row at a time.
+            fmt = ('<tr><td BGCOLOR="{}" BORDER="0" ALIGN="left" '
+                   'COLSPAN="{}">{}</td></tr>')
+            for l in sliced_lines[1:]:
+
+                # Drop LLVM debug call entries
+                if dbg_value.match(l):
+                    continue
+                # find the debug marker but only if interleaving of some form
+                # is requested
+                matched = None
+                if _interleave.lineinfo or _interleave.python:
+                    matched = metadata_marker.match(l)
+                if matched is not None:
+                    # there's a metadata marker
+                    g = matched.groups()
+                    if g is not None:
+                        assert len(g) == 1, g
+                        marker = g[0]
+                        debug_data = md.get(marker, None)
+                        if debug_data is not None:
+                            # and the metadata marker has a corresponding piece
+                            # of metadata
+                            ld = location_entry.match(debug_data)
+                            if ld is not None:
+                                # and the metadata is line info... proceed
+                                assert len(ld.groups()) == 2, ld
+                                line, col = ld.groups()
+                                # only emit a new marker if the line number in
+                                # the metadata is "new".
+                                if line != cur_line or col != cur_col:
+                                    if _interleave.lineinfo:
+                                        mfmt = 'Marker %s, Line %s, column %s'
+                                        mark_line =  mfmt % (marker, line, col)
+                                        ln = fmt.format(cs['marker'], col_span,
+                                                        clean(mark_line))
+                                        new_lines.append(ln)
+                                    if _interleave.python:
+                                        # TODO:
+                                        # +1 for decorator, this probably needs
+                                        # the same thing doing as for the
+                                        # error messages where the decorator
+                                        # is scanned for, its not always +1!
+                                        lidx = int(line) - (firstlineno + 1)
+                                        source_line = src_code[lidx]
```

Comment:
There's a off-by-one error. 
```suggestion
                                        source_line = src_code[lidx + 1]
```